File: JANUARY/24-01-2026/1877.rb
# @param {Integer[]} nums
# @return {Integer}
def min_pair_sum(nums)
  nums.sort!
  i = 0
  j = nums.length - 1
  ans = 0
  while i < j
    ans = [ans, nums[i] + nums[j]].max
    i += 1
    j -= 1
  end
  ans
end

File: JANUARY/24-01-2026/1877.kt
class Solution {
    fun minPairSum(nums: IntArray): Int {
        nums.sort()
        var i = 0
        var j = nums.size - 1
        var ans = 0
        while (i < j) {
            ans = maxOf(ans, nums[i] + nums[j])
            i++
            j--
        }
        return ans
    }
}